src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -191,11 +191,11 @@ JSON file. See [Common metadata fields](#common-metadata-fields) for a
+| RepetitionTimeExcitation | OPTIONAL             | [number][] or [array][] of [numbers][] | The time in seconds between successive excitation pulses that excite a spin system. A single value applies to all volumes, or an array provides a value for each volume. The DICOM tag that best refers to this parameter is [(0018, 0080)](http://dicomlookup.com/lookup.asp?sw=Tnumber&q=(0018,0080)). This field may be used together with the `RepetitionTimePreparation` for certain use cases, such as [MP2RAGE](https://infoscience.epfl.ch/record/172927/files/mp2rage.pdf). Use `RepetitionTimeExcitation` (in combination with `RepetitionTimePreparation` if needed) for anatomy imaging data rather than `RepetitionTime` as it is already defined as the amount of time that it takes to acquire a single volume in the [task imaging data](#task-including-resting-state-imaging-data) section. 4D files with variable `RepetitionTimeExcitation` should only be used in volume timing is critical for interpretation of the data, as in ASL. |
+| RepetitionTimePreparation | OPTIONAL            | [number][] or [array][] of [numbers][] | The period of time in seconds that it takes a preparation pulse block to re-appear at the beginning of the succeeding (essentially identical) pulse sequence block. A single value applies to all volumes, or an array provides a value for each volume. 4D files with variable `RepetitionTimePreparation` should only be used in volume timing is critical for interpretation of the data, as in ASL.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
```

Comment:
```suggestion
| RepetitionTimePreparation | OPTIONAL            | [number][] or [array][] of [numbers][] | The period of time in seconds that it takes a preparation pulse block to re-appear at the beginning of the succeeding (essentially identical) pulse sequence block. The data type number may apply to files from any MRI modality concerned with a single value for this field. The data type array provides a value for each volume in a 4D dataset and should only be used when the volume timing is critical for interpretation of the data, such as in [ASL](01-magnetic-resonance-imaging-data.md#arterial-spin-labeling-perfusion-data).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```

Reply:
When `number` and `array of numbers` are mentioned together, we should be specific about which cases fall into which scope. In the case of `RepetitionTimePreparation` it is `number` except for 4D time-resolved data: 
* The data type number may apply to files from any MRI modality concerned with a single value for this field.
* The data type array provides a value for each volume in a 4D dataset and should only be used when the volume timing is critical for interpretation of the data, such as in ASL. 

It is important that we follow a similar logic, especially for the common fields such as `FlipAngle` and `EchoTime`, where file-collections are scoped as well. To maintain a coherent language, @patsycle agreed to revise the (bep005-revised) definitions of `FlipAngle` and `EchoTime`. For these cases:

* The data type number may apply to files from any MRI modality concerned with a single value for this field.
* The data type number may apply to the files in a file collection where the value of this field is iterated using the _corresponding_ entity.
* The data type array provides a value for each volume in a 4D dataset and should only be used when the volume timing is critical for interpretation of the data, such as in ASL or variable flip angle fMRI sequences. 

My suggestions to the text in BEP005 would then be: 

`FlipAngle`

```diff
- 4D files with variable FA should only be used when volume timing is critical 
- for interpretation of the data, as in ASL or variable FA fMRI sequences.
+ The data type number may apply to files from any MRI modality concerned 
+ with a single value for this field or to the files in a [file collection](./99-appendices/10-file-collections.md)
+ where the value of this field is iterated using the [flip entity](./99-appendices/09-entities.md#flip). 
+ The data type array provides a value for each volume in a 4D dataset and should only be used 
+ when the volume timing is critical for interpretation of the data, such as in [ASL](#arterial-spin-labeling-perfusion-data) or variable flip angle fMRI sequences. 
```

`EchoTime`

```diff
- 4D files with variable FA should only be used when volume timing is critical 
- for interpretation of the data, as in ASL or variable FA fMRI sequences.
+ The data type number may apply to files from any MRI modality concerned 
+ with a single value for this field or to the files in a [file collection](./99-appendices/10-file-collections.md)
+ where the value of this field is iterated using the [echo entity](./99-appendices/09-entities.md#echo). 
+ The data type array provides a value for each volume in a 4D dataset and should only be used 
+ when the volume timing is critical for interpretation of the data, such as in [ASL](#arterial-spin-labeling-perfusion-data) or variable flip angle fMRI sequences. 
```

P.S. I suggest changing **variable FA fMRI** to **variable flip angle fMRI**, as it is agreed that the use of FA can be confusing.

Reply:
Looks good. Just wondering if you should add the link to file collections in the explanation of RepetitionTimePreparation as well?